rules/app.bzl

```diff
@@ -12,6 +13,36 @@ _IOS_APPLICATION_KWARGS = [
     "launch_storyboard",
 ]
 
+def write_info_plists_if_needed(name, plists):
+    """
+    Writes info plists for an app if needed.
+
+    Given a list of infoplists, will write out any plists that are passed as a
+    dict, and will add a default app Info.plist if no non-dict plists are passed.
+
+    Args:
+        name: The name of the app target these infoplists are for.
+        plists: A list of either labels or dicts.
+    """
+    already_written_plists = []
+    written_plists = []
+    for idx, plist in enumerate(plists):
+        if types.is_dict(plist):
+            plist_name = "{name}.infoplist.{idx}".format(name = name, idx = idx)
+            write_file(
+                name = plist_name,
+                destination = plist_name + ".plist",
+                content = struct(**plist).to_json(),
+            )
+            written_plists.append(plist_name)
+        else:
+            already_written_plists.append(plist)
+
+    if not already_written_plists:
+        already_written_plists.append("@build_bazel_rules_ios//rules/test_host_app:Info.plist")
```

Comment:
Does this need to be added in the case that a dict is supplied?

Reply:
Yes, to match cocoapods' behavior